### PR TITLE
Validate https_spiffe profile field trust domains

### PR DESCRIPTION
**Pull request by azdagron**
Relationships using self-serving endpoints have a bundle specified to bootstrap the relationship. In this kind of profile both the bundle and endpoint SPIFFE ID must match the trust domain of relationship.

Relationships using non self-serving endpoints are assumed to receive the bundle to validate the endpoint via seperate configuration. They therefore do not specify the bundle. The endpoint SPIFFE ID must NOT match the trust domain of the trust domain relationship, since it is implied that the endpoint is being served over a different trust domain.

Relevant spec text here:
https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Federation.md#5222-endpoint-parameters

I also fixed up the audit fields being produced for bundles which were not namespaced and in the case of the trust domain ID, were overwriting the field set for the trust domain ID of the relationship itself.

### CODE REVIEW

**Comment by azdagron** [comment]
Not quite ready. I need to add a test to create a non self-serving https_spiffe profile. I think there are bugs there.

File: pkg/server/api/trustdomain.go [hunk 1]
```diff
@@ -63,9 +64,27 @@ func ProtoToFederationRelationshipWithMask(f *types.FederationRelationship, mask
...
+				case trustDomain.IDString() != bundle.TrustDomainId:
+					return nil, fmt.Errorf("self-serving bundle endpoint for %q has a bundle from another trust domain (%q)", trustDomain, profile.HttpsSpiffe.Bundle.TrustDomain)
+				case trustDomain != spiffeID.TrustDomain():
+					return nil, fmt.Errorf("self-serving bundle endpoint for %q has a SPIFFE ID from another trust domain (%q)", trustDomain, spiffeID.TrustDomain())
+				}
+			} else if trustDomain == spiffeID.TrustDomain() {
+				// No bundle means that this is a relationship with a non
+				// self-serving endpoint. The endpoint SPIFFE ID should be
+				// from a different trust domain.
+				return nil, fmt.Errorf("non self-serving bundle endpoint for %q has a SPIFFE ID from the same trust domain", trustDomain)
```
Comment (amartinezfayo) [inline pkg/server/api/trustdomain.go:87]: It would be great if we can cover these cases in the tests also.

Comment (azdagron) [reply pkg/server/api/trustdomain.go:87]: I think we do? check for the tests with  "non self-serving" in the name that were added.

Comment (amartinezfayo) [reply pkg/server/api/trustdomain.go:87]: Right! I was looking at the wrong place.

**Review by amartinezfayo — APPROVED** [review]
LGTM!

**Review by amartinezfayo — APPROVED** [review]

**Pull request merged**